satpy/tests/reader_tests/test_eps_sterna_mwr_l1b.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, 2025 Satpy developers
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for the EPS-Sterna MWR l1b filehandlers."""
+
+from enum import Enum
+
+import numpy as np
+import pytest
+
+from satpy.tests.reader_tests.conftest import make_fake_mwr_lonlats
+
+geo_dims = ["n_scans", "n_fovs", "n_feedhorns"]
+geo_size = 10*145*4
+shape = (10, 145, 4)
+fake_lon_data, fake_lat_data = make_fake_mwr_lonlats(geo_size, geo_dims, shape)
+
+
+@pytest.mark.parametrize(("id_name", "file_key", "fake_array"),
+                         [("longitude", "data/navigation/longitude", fake_lon_data * 1e-4),
+                          ("latitude", "data/navigation/latitude", fake_lat_data),
+                          ])
+def test_get_navigation_data(eps_sterna_mwr_handler, id_name, file_key, fake_array):
+    """Test retrieving the geolocation (lon-lat) data."""
+    Horn = Enum("Horn", ["1", "2", "3", "4"])
```

Comment:
Should this be something that is importable from the reader module so users could do the same thing you're doing here and pass it to a DataQuery/DataID that is then given to `Scene.load`?

Reply:
How do you mean? So, that you could get the lon,lat for a particular horn!?
Geolocation data are a 3-dimensional array, of nscans, nfovs, nhorns...

Reply:
I mean, as a user, what if I wanted to do:

```python
from satpy.readers.mrw_l1b import Horn

scn = Scene(...)
scn.load([DataQuery(name="1", horn=Horn("1"))])
```

But now that I look and think about it I notice a couple things:

1. I think the `Horn` class if done correctly (and satpy itself) the user should be able to pass `horn="1"` and it'll get cast to the Horn Enum automatically. Might be worth testing if not explicitly in a test then just manually as a sanity check. Except...
2. It looks like no dataset/variable has more than one horn? In that sense, I guess they are all unicorns.

Reply:
Yeah, loading channel 1, should apply to horn number 1. There are four receivers/horns, and thus four channel groups.
Horn/Group 1: channel 1-8 (~55 GHz temperature channels)
Horn/Group 2: channel 9 (~89 GHz window channel)
Horn/Group 3: channel 10-15 (~183 GHz humidity channels)
Horn/Group 4: channel 16-19 (~325 GHz humidity channels)

Reply:
Ok so I guess the only real use case would be `scn.load([DataQuery(horn="1")])` but I'm not really sure we even want to support that or how that would work. I mean, I think that usage of a single DataQuery would only return a single channel anyway. Ok so let's ignore this comment/suggestion then.